pyuvdata/utils.py

```diff
@@ -1518,27 +1529,258 @@ def uvcalibrate(
+        ]
+    )
+
+    ant_arr_match = uvcal_used_antnames.tolist() == uvdata_used_antnames.tolist()
+
+    if not ant_arr_match:
+        # check more carefully
+        for this_ant_name in uvdata_used_antnames:
+            wh_ant_match = np.nonzero(uvcal_used_antnames == this_ant_name)
+            if wh_ant_match[0].size == 0:
+                # old behavior only required that antenna numbers were present,#
+                # not names. Check numbers, if they are present issue
+                # deprecation warning
+                uvdata_ant_num = uvdata.antenna_numbers[
+                    np.where(uvdata.antenna_names == this_ant_name)[0][0]
+                ]
+                if override_ant_check:
+                    warnings.warn(
+                        f"Antenna {this_ant_name} has data on UVData but not on UVCal. "
+                        "override_ant_check is True, so the data associated with "
+                        "this antenna will be flagged."
+                    )
+                elif uvdata_ant_num in uvcal_unique_nums:
+                    warnings.warn(
+                        f"Antenna {this_ant_name} has data on UVData but not on UVCal. "
+                        f"Its antenna number is {uvdata_ant_num}, which is present "
+                        "on UVCal. Currently the data will be calibrated using the "
+                        "matching antenna number, but that will be deprecated in "
+                        "version 2.2 and this will become an error. "
+                        "Set override_ant_check=True to proceed "
+                        "with calibration and flag the data for this antenna. ",
+                        DeprecationWarning,
+                    )
+                else:
+                    warn_str = (
+                        f"Antenna {this_ant_name} has data on UVData but not on UVCal. "
+                        "Data for this antenna will not be calibrated"
+                    )
+                    if flag_missing:
+                        warn_str += (
+                            " and will be flagged because flag_missing is True. "
+                            "The flag_missing keyword will be deprected in "
+                            "version 2.2, use override_ant_check instead."
+                        )
+                    else:
+                        warn_str += (
+                            ", set override_ant_check=True to proceed "
+                            "with calibration and flag the data for this antenna. "
+                            "Currently calibration will proceed, the data will not "
+                            "be flagged and not be calibrated. This behavior will "
+                            "deprected in version 2.2."
+                        )
+                    warnings.warn(warn_str, DeprecationWarning)
+
+    uvdata_times = np.unique(uvdata.time_array)
+    downselect_cal_times = False
+    if uvcal.Ntimes > 1:
```

Comment:
I'm wondering if it is also useful to make this `if uvcal.Ntimes > 1 and not override_time_check:`
the benefit is that it will entirely skip this next block if one wants to override the time check. This is nice b/c in my experience, some of these metadata checks can become take quite a while (the time check on uvdata for example was actually a non-negligible bottleneck in loading night-long uvdata waterfalls).

Reply:
Same for the above ant_check, although not sure if thats as much of a memory / wall-time concern

Reply:
No. We only allow a time override in the case that you have a single solution (Ntimes = 1) that you want to apply to a different time. Otherwise you have an array of solutions for each time integration and it doesn't really make sense to apply that to another unrelated data set -- first, it might have a different number of time integrations, second why would you want exactly the same sequence of solutions applied to an unrelated file? So the `override_time_check` is ignored if Ntimes > 1.

Reply:
I see, but my concern isn't so much about the physical reason and more about the computation required to do the check. Like I said with UVData, doing time checks across a nightly waterfall turned out to be a sizeable fraction of the load time, and it was helpful to be able to skip it if desired. 

Seems like there may also be good reason to be able to manually skip the checks (as is done in UVData to my understanding) in case one is doing something experimental. Of course they would be by default True so most anyone would be running them anyways.

Reply:
Ok, we'll talk about it on the telecon tomorrow. You're welcome to join you like!

Reply:
The feeling on the telecon is that you only end up in the slow part of the check (starting on line 1633 below) in very strange edge cases. If your UVCal times are the same as your UVData times then it's just an `np.allclose` call which is fast. If you have a single UVCal time or just a UVCal range you don't enter this slow part of the check.

Reply:
I see, that makes sense to me